### PR TITLE
[core] FileBasedBloomFilter should not be pass private member to cache manager.

### PR DESCRIPTION
**Pull request by leaves12138**
### Purpose

Cache manager cause out of memory while lookup merge

### CODE REVIEW

File: paimon-common/src/main/java/org/apache/paimon/utils/FileBasedBloomFilter.java [hunk 1]
```diff
@@ -58,12 +58,13 @@ public boolean testHash(int hash) {
...
         // so we introduce a refresh count to reduce refresh
         if (accessCount == REFRESH_COUNT || filter.getMemorySegment() == null) {
+            final BloomFilter bloomFilter = filter;
             MemorySegment segment =
                     cacheManager.getPage(
                             CacheKey.forPosition(input.file(), readOffset, readLength),
                             key -> input.readPosition(readOffset, readLength),
-                            key -> filter.unsetMemorySegment());
-            filter.setMemorySegment(segment, 0);
+                            key -> bloomFilter.unsetMemorySegment());
```
Comment (JingsongLi) [inline paimon-common/src/main/java/org/apache/paimon/utils/FileBasedBloomFilter.java:66]: add a static inner class to do this unset.

**Comment by JingsongLi** [comment]
+1

**Comment by Aitozi** [comment]
Hi @leaves12138 , why this will lead to out of memory ?

**Comment by leaves12138** [comment]
> Hi @leaves12138 , why this will lead to out of memory ?

If we use lambda, the member "filter" will be indexed. The instance which contains the "filter" will also be referred, cannot gc

**Pull request merged**